resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -458,6 +462,51 @@ private[spark] class Client(
     new Path(resolvedDestDir, qualifiedDestPath.getName())
   }
 
+  /**
+   * For each non-local and non-glob resource, we will count its parent directory. If its
+   * frequency is larger than the threshold specified by
+   * spark.yarn.client.statCache.preloaded.perDirectoryThreshold, all the file status from the
+   * directory will be preloaded.
+   *
+   * @param jars : the list of jars to upload
+   * @return a list of directories to be preloaded
+   * */
+  def directoriesToBePreloaded(jars: Seq[String]): mutable.Iterable[URI] = {
```

Comment:
Maybe return a `Seq` or `List`? Even your Scaladoc says you're returning a list :) `Iterable` has some caveats like potentially only being iterable a single time, I generally wouldn't recommend passing it around unless you have a need to (e.g. you're generating elements on the fly and don't want to materialize the whole thing)

also, this should be `private[yarn]` ?